lib/persisted-model.js

```diff
@@ -1488,12 +1488,9 @@ module.exports = function(registry) {
   }
 
   function getIdFromWhereByModelId(Model, where) {
-    var whereKeys = Object.keys(where);
-    if (whereKeys.length != 1) return undefined;
-
     var idName = Model.getIdName();
-    if (whereKeys[0] !== idName) return undefined;
-
+    if (!(idName in where))
+      return undefined;
```

Comment:
The original code used empty lines to visually delimit blocks of code (sub-steps). Please don't remove them!